.github/workflows/any-pr.yml

```diff
@@ -21,12 +21,10 @@ jobs:
       # Logging python version for future debugging
       - run: python -V
 
-      # Required as macos-14 defaults to use python 3.12, this causes issues with node-gyp currently
-      # 1. node-gyp depends on distutils package which has now been removed in python version 3.12 can use 'pip install setuptools' to fix this
-      # 2. Node@14 will error with -> Node gyp ERR - invalid mode: 'rU' while trying to load binding.gyp Where 'U' option is now deprecated
       - uses: actions/setup-python@v4
         with:
-          python-version: "3.10.11"
+          python-version: "3.12"
+      - run: pip install setuptools
```

Comment:
Just thinking, do we use setuptools anywhere else in other projects? It would be like a new dev dependency right? If yes and we don't use it anywhere else we'd probably need to onboard it as a new dependency and ask for approvals

Reply:
We use setuptools in [mattrglobal/ffi-bbs-signatures](https://github.com/mattrglobal/ffi-bbs-signatures) [.github/workflows/ci-python.yml](https://github.com/mattrglobal/ffi-bbs-signatures/blob/ee979624ac1bff78adaac60bd6b0ec546cc73413/.github/workflows/ci-python.yml#L126). It is also used in [mattrinternal/universal-resolver](https://github.com/mattrinternal/universal-resolver) [ci/deploy-k8s-aws/Dockerfile](https://github.com/mattrinternal/universal-resolver/blob/960e7c033b3fd88cd0c6912f51497421c885a4a0/ci/deploy-k8s-aws/Dockerfile#L20)